warehouse/api/echo.py

```diff
@@ -79,34 +79,15 @@ def api_echo(request: Request):
     permission=Permissions.APIObservationsAdd,
     require_methods=["POST"],
 )
-def api_projects_observations(
-    project: Project, request: Request
-) -> HTTPAccepted | HTTPBadRequest:
+def api_projects_observations(project: Project, request: Request) -> dict:
     data = request.json_body
 
-    # TODO: Are there better mechanisms for validating the payload?
```

Comment:
These removals and their associated test case removals are now externalized to the "contract" of the OpenAPI document.